code/__HELPERS/unsorted.dm

```diff
@@ -1501,6 +1501,17 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 
 #define UNTIL(X) while(!(X)) stoplag()
 
+#define UNTIL_OR_TIMEOUT(X, __time) \
```

Comment:
__time also might better if it was forced to be >= 0 (e.g. with max it and 0) else the timeout will never possibly occur.

Reply:
done